types/scheduler.d.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Task priorities that determine the order in which tasks run.
+ *
+ * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Prioritized_Task_Scheduling_API#task_priorities)
+ */
+export type TaskPriority = 'user-blocking' | 'user-visible' | 'background';
+
+/**
+ * {@link Scheduler.postTask} options.
+ *
+ * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Scheduler/postTask#options)
+ */
+export type SchedulerPostTaskOptions = {
+  /** The immutable {@link TaskPriority} of the task. One of `"user-blocking"`, `"user-visible"`, or `"background"`. If set, this priority is used for the lifetime of the task and priority set on the `signal` is ignored. */
+  priority?: TaskPriority;
+  /** An {@link AbortSignal} or {@link TaskSignal} that can be used to abort or re-prioritize the task (from its associated controller). */
+  signal?: AbortSignal | TaskSignal;
```

Comment:
Is `| TaskSignal` needed since `TaskSignal` extends `AbortSignal`? It wasn't for the IDL, but not sure how that works for TS. Either way, mentioning `TaskSignal` makes sense for documentation since the signal's priority is used for the task if no fixed `priority` was provided.

Reply:
> Is `| TaskSignal` needed since `TaskSignal` extends `AbortSignal`? It wasn't for the IDL, but not sure how that works for TS. Either way, mentioning `TaskSignal` makes sense for documentation since the signal's priority is used for the task if no fixed `priority` was provided.

Yeah, it's the same. To the caller of `postTask`/`yield` it could just be `AbortSignal` and there would be no difference in functionality, so it's more or less just documentation.

(I was also prototyping updating the jsdoc in the polyfill to be type checkable by the typescript compiler, and there it's useful so that the check for `priority` can narrow the type to `TaskSignal` without downcasting, but I'm not sure if adding all the extra types will be worth it)